service/frontend/handler.go

```diff
@@ -1029,31 +1020,50 @@ func (wh *WorkflowHandler) GetWorkflowExecutionHistory(
 		token.RunID = runID
 		token.FirstEventID = 0
 		token.NextEventID = nextEventID
-		token.IsWorkflowRunning = isRunning
+		token.IsWorkflowRunning = isWorkflowRunning
 		token.PersistenceToken = nil
 	}
 
 	history := &gen.History{}
-	if token.FirstEventID >= token.NextEventID {
-		// currently there is no new event
-		events := []*gen.HistoryEvent{}
-		history.Events = events
-		if !token.IsWorkflowRunning {
+	history.Events = []*gen.HistoryEvent{}
+	if isCloseEventOnly {
```

Comment:
Probably it makes sense to push the filter all the way down to history service.

Reply:
ok

Reply:
done using the int64 max as expected next event ID